### DISCUSSION TITLE
Plot IPF with labels

### DISCUSSION THREAD

**Discussion by JOchoaA** [mtex-toolbox/mtex#1866] [Ask Anything]
Dear all,

First of all, thanks for this amazing software.

I have two lists of n orientations (deformed and undeformed conditions). I want to plot them as in the picture attached to see their rotation trajectories in an IPF.  I managed to plot them, but now I don't know how to label them, have a colour for undeformed and another for deformed and the most difficult, to add the line representing the trajectory. 

Thanks for any help, and thanks for your patience.

Cheers

Jhon Ochoa

![image](https://github.com/mtex-toolbox/mtex/assets/61281507/21e7f0bc-99b9-45d6-90e4-4cfefdf232f3)

**Comment by kilir** [comment]
Hi,

how about that:

```
%some data
cs = crystalSymmetry('321')
n=5
h1 = Miller.rand(n,cs)
h1 = h1.project2FundamentalRegion;
rot = rotation.byAxisAngle(vector3d.rand(n),rand(n,1));
h2 = rot .* h1
h2 = h2.project2FundamentalRegion;

plot(h1,'fundamentalRegion');
text(h1,arrayfun(@num2str,1:n,'uniformOutput',false))
hold on
plot(h2,'fundamentalRegion');
line(vector3d(h1),vector3d(h2))
hold off
```
Cheers,
Rüdiger

**Comment by JOchoaA** [comment]
Hi Rüdiger,

Thanks for your reply. It does exactly what I need, but I got an additional issue. My orientations are in Euler angles, so I used plotIPDF( ) function.  

If I use only plot( ), I get the plot in Euler space and lose both the label and the lines connecting the orientations

any idea of how to solve it?

Thanks in advance.

%%
cs = crystalSymmetry('cubic');
n=1;
ori_euler = [35, 70, 219];
rot = rotation.byAxisAngle(vector3d.rand(n),rand(n,1));
h1 = orientation.byEuler(ori_euler(1)*degree,ori_euler(2)*degree,ori_euler(3)*degree,cs);
h1 = h1.project2FundamentalRegion;
h2 = rot .* h1;
h2 = h2.project2FundamentalRegion;

plotIPDF(h1,vector3d.X,'fundamentalRegion');
% text(h1,arrayfun(@num2str,1:n,'uniformOutput',false))
hold on
plotIPDF(h2,vector3d.X,'fundamentalRegion');
line(vector3d(h1),vector3d(h2))
hold off

**Comment by kilir** [comment]
In that case, remember that h = inv(o) * r. 
e.g.
```
n=5
o1 = orientation.rand(n,cs)
rot = rotation.byAxisAngle(vector3d.rand(n),rand(n,1));
o2 = rot .* o1

% compute crystakl directions
ref = vector3d(19,2,1)

h1 = inv(o1) * ref;
h1 = h1.project2FundamentalRegion;
h2 = inv(o2) * ref;
h2 = h2.project2FundamentalRegion;
```

**Comment by JOchoaA** [comment]
hmm, that's true!

Problem solved! 

Thanks for your help Rüdiger,

Best wishes,

Jhon
